src/commons/utils/InfiniteLoopReporter.ts

```diff
@@ -0,0 +1,162 @@
+import * as Sentry from '@sentry/browser';
+import { Context, NativeStorage, SourceError } from 'js-slang/dist/types';
+
+/**
+ * Parses the error and determines whether it is an infinite loop and
+ * if so, what kind of infinite loop.
+ *
+ * @returns {string} string containing the infinite loop classification.
+ * @returns {string} empty string, if the error is not an infinite loop.
+ */
+function infiniteLoopErrorType(errorString: string): string {
+  if (errorString.includes('Infinite recursion (or runtime error) detected')) {
+    if (errorString.includes('forget your base case')) return 'no_base_case';
+    else if (
+      errorString.includes('Did you call a value that is outside the range of your function?')
+    )
+      return 'input_out_of_domain';
+    else if (errorString.includes('Check your recursive function calls.')) return 'no_state_change';
+  }
```

Comment:
Might be better to have a special error type enum where these errors are thrown (in js-slang) so that this won't break if someone changes the error message wording

Reply:
ok I'll work on adding this to js-slang as an enum, same with L27-30.